docs/cli-roadmap.md

```diff
@@ -0,0 +1,82 @@
+# Kubernetes CLI/Configuration Roadmap
+
+See also issues with the following labels:
+* [area/config-deployment](https://github.com/GoogleCloudPlatform/kubernetes/labels/area%2Fconfig-deployment)
+* [component/CLI](https://github.com/GoogleCloudPlatform/kubernetes/labels/component%2FCLI)
+* [component/client](https://github.com/GoogleCloudPlatform/kubernetes/labels/component%2Fclient)
+
+At least the first three issues should block 1.0.
+
+1. stop command in kubectl [#2144](https://github.com/GoogleCloudPlatform/kubernetes/issues/2144): Gracefully terminate any object(s), to the extent currently possible. We also eventually want first-class support in the API for graceful termination. [#1535](https://github.com/GoogleCloudPlatform/kubernetes/issues/1535)
+1. Kind-based filtering on object streams -- only operate on the kinds of objects specified
+1. Create services before other objects, or at least before objects that depend upon them. Namespace-relative DNS mitigates this some, but most users are still using service environment variables. [#1768](https://github.com/GoogleCloudPlatform/kubernetes/issues/1768)
+1. Pretty printing of endpoints
+1. Service address/port lookup command(s)
+1. Multiple objects on command line [#3050](https://github.com/GoogleCloudPlatform/kubernetes/issues/3050)
+1. Finish rolling update [#1353](https://github.com/GoogleCloudPlatform/kubernetes/issues/1353)
+  1. Friendly to auto-scaling [#2863](https://github.com/GoogleCloudPlatform/kubernetes/pull/2863#issuecomment-69701562)
+  1. Rollback
+  1. Rollover
+  1. Write a ReplicationController generator to derive the new ReplicationController from an old one
+  1. Use readiness [#620](https://github.com/GoogleCloudPlatform/kubernetes/issues/620)
+  1. Perhaps factor this in a way that it can be shared with [Openshift’s deployment controller](https://github.com/GoogleCloudPlatform/kubernetes/issues/1743) 
+  1. Rolling update service as a plugin
+1. List supported API versions
+1. List supported resources
+1. Swagger lookups [#3060](https://github.com/GoogleCloudPlatform/kubernetes/issues/3060)
+1. --name, --name-suffix applied during creation and updates
+1. --labels and opinionated label injection: --app=foo, --tier={fe,cache,be,db}, --uservice=redis, --env={dev,test,prod}, --stage={canary,final}, --track={hourly,daily,weekly}, --release=0.4.3c2. Exact ones TBD. We could allow arbitrary values -- the keys are important. The actual label keys would be (optionally?) namespaced with kubectl.kubernetes.io/, or perhaps the user’s namespace.
```

Comment:
More ideas, based in part on #3576:
* We could support something like --image-version=foo, which would append a name suffix, create a label, and apply a tag to the image. That could decompose cleanly into several independent options (--name-suffix, label options, and a tag option).
* We might also want a --registry option.